QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java

```diff
@@ -61,18 +58,18 @@ public class QueryDAO {
 
 	public Uni<Map<String, Object>> getEntity(String entityId, String tenantId, AttrsQueryTerm attrsQuery) {
 		return clientManager.getClient(tenantId, false).onItem().transformToUni(client -> {
-			String sql = "";
-			sql += "SELECT ENTITY";
-
-			Tuple tuple = Tuple.tuple();
+			String sql = "SELECT ";
 			if (attrsQuery != null) {
-				sql += "-$1 FROM ENTITY WHERE ID=$2";
-				tuple.addArrayOfString(attrsQuery.getAttrs().toArray(new String[0]));
+				for (String attrs : attrsQuery.getAttrs()) {
+					sql += "JSONB_BUILD_OBJECT('%s', ENTITY->'%s') ||";
+					sql = sql.formatted(attrs, attrs);
+				}
+				sql = sql.substring(0, sql.length() - 2);
 			} else {
-				sql += " FROM ENTITY WHERE ID=$1";
+				sql += "ENTITY ";
 			}
-			tuple.addString(entityId);
-			return client.preparedQuery(sql).execute(tuple).onItem().transformToUni(t -> {
+			sql += " FROM ENTITY WHERE ID='%s'".formatted(entityId);
```

Comment:
in general use preparequeries properly with dollar symbols.
formatted does not sql escape. execute(tuple) does.

Reply:
done